component/models/dataintegrity/signer.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright Gen Digital Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package dataintegrity
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/tidwall/sjson"
+
+	"github.com/hyperledger/aries-framework-go/component/models/dataintegrity/models"
+	"github.com/hyperledger/aries-framework-go/component/models/dataintegrity/suite"
+)
+
+// Signer implements the Add Proof algorithm of the verifiable credential data
+// integrity specification, using a set of provided cryptographic suites.
+type Signer struct {
+	suites map[string]suite.Signer
+}
+
+// NewSigner initializes a Signer that supports using the provided cryptographic
+// suites to perform data integrity signing.
+func NewSigner(suites ...suite.SignerInitializer) (*Signer, error) {
+	signer := &Signer{
+		suites: map[string]suite.Signer{},
+	}
+
+	for _, initializer := range suites {
+		suiteType := initializer.Type()
+
+		if _, ok := signer.suites[suiteType]; ok {
```

Comment:
Is this check here to ensure there aren't duplicates? I'm wondering if maybe you should state this in the NewSigner comment docs or perhaps make the input argument a map to begin with? Or maybe that makes the parameter too complicated.... hmm. Something to think about

Reply:
Yes, essentially. Without a check, then multiple instances of the same suite might be initialized (if, for example, you have an API that initializes some defaults and then you also pass in a list of suites you want to include, they might overlap), but it doesn't make sense to have multiple versions of the same suite active in the same data-integrity signer/verifier. So, we'd still only have one instance of each suite available in the signer/verifier, meaning all the others were initialized for no reason. Since initialization might mean initializing some sort of key store etc, I thought it would be better to avoid letting duplicates initialize in the first place.